jvm/library/src/main/kotlin/io/tira/tirex/tracker/Tracker.kt

```diff
@@ -763,8 +788,28 @@ class TrackingHandle private constructor(
             javaInfo.getValue(Measure.JAVA_SPECIFICATION_NAME).value?.let { json.decodeFromString<String?>(it) }
 
         // Serialize the updated ir_metadata.
-        exportFilePath.outputStream().bufferedWriter().use { writer ->
+        val stream = exportFilePath.outputStream().let {
+            if (exportFilePath.name.endsWith(".gz")) {
```

Comment:
Just a note: if we were to use a case-insensitive regex above, then this must also be case insensitive

Reply:
True. Let's keep both case-sensitive.